src/js/veteran-id-card/containers/VeteranIDCard.jsx

```diff
@@ -7,33 +7,39 @@ import EmailCapture from './EmailCapture';
 
 const rateLimitAuthed = window.settings.vic.rateLimitAuthed;
 const rateLimitUnauthed = window.settings.vic.rateLimitUnauthed;
+const serviceRateLimitedRandomizer = Math.random();
 
 class VeteranIDCard extends React.Component {
 
-  componentDidMount() {
-    // Redirect users to email form based on rate limits for unauthed or authed users
-    if (this.props.profile.accountType) {
-      if (this.props.serviceRateLimitedAuthed) {
-        window.dataLayer.push({ event: 'vic-authenticated-ratelimited' });
-        this.renderEmailCapture = true;
-      } else {
-        window.dataLayer.push({ event: 'vic-authenticated' });
-      }
-    } else {
-      if (this.props.serviceRateLimitedUnauthed) {
-        window.dataLayer.push({ event: 'vic-unauthenticated-ratelimited' });
-        this.renderEmailCapture = true;
-      } else {
-        window.dataLayer.push({ event: 'vic-unauthenticated' });
-      }
-    }
+  componentWillReceiveProps(nextProps) {
+
+    // Once the login logic is all done...
+    // This will occur even for unauthenticated users and should only occur once.
+    if (this.props.profile.loading && !nextProps.profile.loading) {
+      const userProfile = nextProps.profile;
+      const serviceRateLimitedAuthed = this.props.serviceRateLimitedRandomizer > this.props.rateLimitAuthed;
+      const serviceRateLimitedUnauthed = this.props.serviceRateLimitedRandomizer > this.props.rateLimitUnauthed;
```

Comment:
Why is the randomizer mapped to props instead of accessing it from the constant defined above?

Reply:
That prop is still mapped to that constant (see `mapStateToProps`). I moved it so that the value is exposed as a prop and so I can see it in the React dev tools. Also it sort of sets it up to move that value into the Redux layer. Our next step is to not apply the `rateLimitAuthed` rate limit to users who logged in after being allowed through by the `rateLimitUnauthed`. I think that logic would probably be best in Redux logic, right?

Reply:
@ncksllvn was wondering the same--is there a reason you moved it into props?

Reply:
Yeah I answered that a moment ago but it may not have come through on your comments yet

> That prop is still mapped to that constant (see mapStateToProps). I moved it so that the value is exposed as a prop and so I can see it in the React dev tools. Also it sort of sets it up to move that value into the Redux layer. Our next step is to not apply the rateLimitAuthed rate limit to users who logged in after being allowed through by the rateLimitUnauthed. I think that logic would probably be best in Redux logic, right?

Reply:
If anyone disagrees, we can pull it out of props. I just don't like having values that we can't see or change from unit testing or parent components.

Reply:
I don't have a strong opinion on props or not; I have thoughts on moving this into Redux (e.g. reducers need to be pure) but we can revisit that later.

Reply:
Yeah I was thinking using these constants as `initialState` and then just never overriding them unless we find ourselves under special circumstances, not sure what those would be right now though